name: CI

on:
  push:
    branches:
      - main
      - 7-authorization-system
  pull_request:
    branches:
      - main

jobs:
  build:
    runs-on: ubuntu-22.04

    steps:
      - name: Checkout code
        uses: actions/checkout@v2

      - name: Set up Python
        uses: actions/setup-python@v2
        with:
          python-version: '3.x'

      - name: Install Python dependencies
        run: |
          python -m pip install --upgrade pip
          pip install -r backend/requirements.txt

      - name: Run Python linter
        run: |
          flake8 backend --config=backend/.flake8
        env:
          PYTHON_LINTER_ERROR: "0"
        continue-on-error: true

      - name: Set up Node.js
        uses: actions/setup-node@v2
        with:
          node-version: '18.x'

      - name: Install Node.js dependencies
        run: cd frontend && npm install

      - name: Run Node.js linter
        run: cd frontend && npm run lint
        env:
          JS_LINTER_ERROR: "0"
        continue-on-error: true

      - name: Check for errors
        run: |
          if [ "$PYTHON_LINTER_ERROR" -eq "1" ] || [ "$JS_LINTER_ERROR" -eq "1" ]; then
            echo "There were errors in the previous steps"
          fi
